app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -458,16 +483,20 @@ protected function _setFilterValues($data)
         return $this;
     }
 
+    /**
+     * @param Mage_Adminhtml_Block_Widget_Grid_Column $column
+     * @return $this
+     */
     protected function _addColumnFilterToCollection($column)
     {
         if ($this->getCollection()) {
-            $field = ( $column->getFilterIndex() ) ? $column->getFilterIndex() : $column->getIndex();
+            $field = $column->getFilterIndex() ?: $column->getIndex();
             if ($column->getFilterConditionCallback() && $column->getFilterConditionCallback()[0] instanceof self) {
                 call_user_func($column->getFilterConditionCallback(), $this->getCollection(), $column);
             } else {
                 $cond = $column->getFilter()->getCondition();
-                if ($field && isset($cond)) {
-                    $this->getCollection()->addFieldToFilter($field , $cond);
+                if ($field) {
```

Comment:
I see. Is this better:

```php
            if ($column->getFilterConditionCallback() && $column->getFilterConditionCallback()[0] instanceof self) {
                call_user_func($column->getFilterConditionCallback(), $this->getCollection(), $column);
            } else {
                $field = $column->getFilterIndex() ?: $column->getIndex();
                if ($field) {
                    $this->getCollection()->addFieldToFilter($field, $column->getFilter()->getCondition());
```

Reply:
Think `else` and `if` could combined ... next PR ....